AmazonChimeSDKDemo/AmazonChimeSDKDemo/CaptionsModel.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  CaptionsModel.swift
+//  AmazonChimeSDKDemo
+//
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+
+import AmazonChimeSDK
+import UIKit
+
+class CaptionsModel: NSObject {
+    let logger = ConsoleLogger(name: "CaptionsModel")
+    
+    var isLiveTranscriptionEnabled = false
+    
+    var captions = [Caption]()
+    var captionIndices = [String: Int]()    // map resultId to index in captions array
+    var refreshCaptionsTableHandler: (() -> Void)?
+
+    // triggered on main thread, no synchronization needed
+    public func addTranscriptEvent(transcriptEvent: TranscriptEvent) {
+        if let status = transcriptEvent as? TranscriptionStatus {
+            let formattedEventTime = TimeStampConversion.formatTimestamp(timestamp: status.eventTimeMs)
+            let content = "Live transcription \(status.type) at \(formattedEventTime) in \(status.transcriptionRegion) with configuration: \(status.transcriptionConfiguration)"
+            let caption = Caption(speakerName: "",
+                                  isPartial: false,
+                                  content: content)
+            captions.append(caption)
+            if status.type == .started || status.type == .stopped {
+                isLiveTranscriptionEnabled = status.type == .started
+            }
+        } else if let transcript = transcriptEvent as? Transcript {
+            transcript.results.forEach { result in
+                guard let alternative = result.alternatives.first else {
+                    return
+                }
+
+                // for simplicity and demo purposes, assume each result only contains transcripts from
+                // the same speaker, which matches our observation with current transcription service behavior.
+                // More complicated UI logic can be achieved by iterating through each item
+                var speakerName: String
+                if let firstItem = alternative.items.first {
+                    speakerName = RosterModel.convertAttendeeName(from: firstItem.attendee)
+                } else {
+                    logger.debug(debugFunction: {
+                        return "Empty speaker name due to empty items array for result: \(result.resultId)"
+                    })
+                    speakerName = ""
```

Comment:
Default `speakerName` to "" when declaration?

Reply:
Defaulted to "<UNKNOWN>" to be consistent with a later change in https://github.com/aws/amazon-chime-sdk-ios/pull/355